### PR TITLE
Add base deps to `pub.yml`

### PR DESCRIPTION
**Pull request by tomvothecoder**
- Update environment instructions in `warehouse.md`
- Replace `yaml` with `pyyaml`

### CODE REVIEW

File: conda-env/pub.yml [hunk 1]
```diff
@@ -13,6 +13,18 @@ dependencies:
   # ==================
   - python=3.8.10
   - pip=22.0.3
+  - pyyaml=6.0
+  - cmor=3.6.1
+  - netcdf4=1.5.6
```
Comment (tomvothecoder) [inline conda-env/pub.yml:18]: `netcdf4=1.5.8` (the latest version) conflicts with `autocurator=0.1`.

Have to fall back to 1.5.6.

**Comment by tomvothecoder** [comment]
Hey @TonyB9000, this PR should fix the `warehouse_pub` missing Python imports.

Can you checkout this branch and try the publishing operation to make sure no other dependencies are missing?

Here are the commands to follow:

```
cd esgfpub
git fetch
git checkout devops/pub-env-deps
conda env create -f conda-env/pub.yml
conda activate warehouse_pub
```
You should also probably delete the existing `warehouse_pub` env that you already created (`conda env remove -n warehouse_pub`).

File: conda-env/pub.yml (hunk 1 above)
Comment (TonyB9000) [reply conda-env/pub.yml:18]: Ok by me.  But Sasha should be pulled on this - not to approve, but just to make aware.

**Review by TonyB9000 — APPROVED** [review]
Excellent.

**Comment by TonyB9000** [comment]
"warehouse publish --help" no longer bombs.  Good sign.  Now we test with a real dataset.

**Pull request merged**